enhancements/upgrade-bundle.md

```diff
@@ -0,0 +1,260 @@
+
+#### Example UX Scenarios
+
+Here are some example scenarios of using `run bundle-upgrade`:
+
+* `operator-sdk run bundle-upgrade <bundle-image> --index-image quay.io/operator-framework/upstream-opm-builder --starting-csv test-operator.v1.0.0`
+
+* `operator-sdk run bundle-upgrade <bundle-image> --index-image quay.io/operator-framework/upstream-opm-builder --starting-csv test-operator.v1.0.0 --channel alpha`
+
+* `operator-sdk run bundle-upgrade <bundle-image> --index-image quay.io/operator-framework/upstream-opm-builder --starting-csv test-operator.v1.0.0 --channel alpha --namespace memcached-system`
+
+* `operator-sdk run bundle-upgrade <bundle-image> --index-image quay.io/operator-framework/upstream-opm-builder --starting-csv test-operator.v1.0.0 --channel alpha --namespace memcached-system --install-mode SingleNamespace=default`
+
+This is how the presence of an OperatorGroup in the
+namespace affects `run bundle-upgrade` behavior. 
+
+* Given an operator that supports install modes `OwnNamespace` and
+  `SingleNamespace` being upgraded into a namespace with *no* OperatorGroup.
+  * `operator-sdk run bundle-upgrade <bundle-image> --index-image quay.io/operator-framework/upstream-opm-builder --starting-csv test-operator.v1.0.0`
+    * Upgrade into the default namespace configured in the `KUBECONFIG` context. An OperatorGroup would be created targeting the same install namespace.
+  * `operator-sdk run bundle-upgrade <bundle-image> --index-image quay.io/operator-framework/upstream-opm-builder --starting-csv test-operator.v1.0.0 --namespace memcached-system --install-mode SingleNamespace=default`
+      * Upgrade into namespace `memcached-system`. An OperatorGroup would be created targeting the `default` namespace.
+* Given an operator that supports install modes `AllNamespaces`, `OwnNamespace`, and `SingleNamespace`, being upgraded into a namespace with an existing OperatorGroup that targets all namespaces:
+  * `operator-sdk run bundle-upgrade <bundle-image> --index-image quay.io/operator-framework/upstream-opm-builder --starting-csv test-operator.v1.0.0` 
+    * Upgrade into the default namespace configured in the `KUBECONFIG` context. The operator would watch all namespaces.
+  * `operator-sdk run bundle-upgrade <bundle-image> --index-image quay.io/operator-framework/upstream-opm-builder --starting-csv test-operator.v1.0.0 --namespace memcached-system --install-mode SingleNamespace=default`
+    * Upgrade into namespace `memcached-system`. The operator would watch namespace `default`.
+
+#### Simple Example
+
+Here's a simple example with all the required flags.
+
+```
+operator-sdk run bundle-upgrade <bundle-image> --index-image quay.io/operator-framework/upstream-opm-builder --starting-csv test-operator.v1.0.0
+```
+
+Below is the workflow when the above command is invoked:
+
+ 1. Create pod using `--index-image`, set container command (entrypoint) to:
+    ```
+    /bin/bash -c ‘ \
```

Comment:
this assumes the index contains `bash`, which may not always be the case.

Reply:
this is one of the assumptions in this proposal that the index image has bash to be able to change the entrypoint. 
Will explicitly state this as an assumption